mint-auditor/src/db/mod.rs

```diff
@@ -163,47 +174,70 @@ impl MintAuditorDb {
             }
 
             // Count burns.
-            for tx_out in &block_contents.outputs {
-                if let Ok((amount, _)) = tx_out.view_key_match(&burn_address_view_private()) {
-                    let balance = balance_map.entry(amount.token_id).or_default();
-
-                    if amount.value > *balance {
-                        log::crit!(
-                            self.logger,
-                            "Block {}: Burned {} of token id {} but only had {}. Setting balance to 0",
-                            block_index,
-                            amount.value,
-                            amount.token_id,
-                            balance
-                        );
-                        *balance = 0;
-                        counters.num_burns_exceeding_balance += 1;
-                    } else {
-                        *balance -= amount.value;
-                        log::info!(
-                            self.logger,
-                            "Block {}: Burned {} of token id {}, balance is now {}",
-                            block_index,
-                            amount.value,
-                            amount.token_id,
-                            balance,
-                        );
-                    }
+            log::trace!(self.logger, "Processing burns");
+            let burn_amounts = block_contents
+                .outputs
+                .par_iter()
+                .filter_map(|tx_out| {
+                    tx_out
+                        .view_key_match(&burn_address_view_private())
```

Comment:
This seems crucial, this ensures only the outputs sent to the burn address get considered right? (I.e. as those are the only amounts we can perform ristretto group algebra on to get an amount.)

Reply:
Right - figured I'd point this subtle detail out for any member of the public who comes upon this PR